staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions.go

```diff
@@ -437,6 +439,22 @@ func (o *EditOptions) Run() error {
 	}
 }
 
+func stripManagedFields(obj runtime.Object) error {
```

Comment:
I think technically this should work, however I am not sure it is future proof if we e.g. add another field format. I would greatly prefer it if you put them back before constructing the patch that will be sent to apiserver.